lib/core/AbstractVersionMetadata.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * Holds metadata for a particular Sidetree version needed by the orchestration layer classes across all versions of the Sidetree.
+ */
+export default abstract class AbstractVersionMetadata {
```

Comment:
Let's create a folder perhaps named `abstracts` similar to `interfaces` and `models` as it is possible that we will slowly migrate to this new pattern.